version: '3.8'

services:
  traefik:
    image: traefik:latest
    restart: always
    ports:
      - '80:80'
      - '443:443'
    profiles:
      - listener
      - server
    volumes:
      - ./traefik:/etc/traefik
      - /var/run/docker.sock:/var/run/docker.sock:ro
  
  configlistener:
    build:
      context: ./config
    stop_signal: SIGINT
    environment:
      - FLASK_SERVER_PORT=9091
    volumes:
      - config:/config
      - venv:/venv
    profiles:
      - listener
    labels:
      - 'traefik.enable=true'
      - 'traefik.http.routers.my-app.rule=Host(`{EXTERNAL_URL}`)'
      - 'traefik.http.routers.my-app.tls=true'
      - 'traefik.http.routers.my-app.tls.certresolver=lets-encrypt'
      - 'traefik.http.services.my-app.loadbalancer.server.port=9091'

  backend:
    build:
      context: ./src
    stop_signal: SIGINT
    environment:
      - FLASK_SERVER_PORT=9091
    volumes:
      - src:/src
      - config:/config
      - venv:/venv
    profiles:
      - server
    labels:
      - 'traefik.enable=true'
      - 'traefik.http.routers.my-app.rule=Host(`{EXTERNAL_URL}`)'
      - 'traefik.http.routers.my-app.tls=true'
      - 'traefik.http.routers.my-app.tls.certresolver=lets-encrypt'
      - 'traefik.http.services.my-app.loadbalancer.server.port=9091'
